tests/unit/test_aws_lambda_handler.py

```diff
@@ -80,3 +125,34 @@ def test_lambda_handler(monkeypatch, apigw_event, tmpdir):
 
     assert ret["statusCode"] == 200
     assert predictions == [8.0, 8.0, 0.0]
+
+
+def test_s3_event_lambda_handler(monkeypatch, features, s3_event, tmpdir, caplog):
+
+    from tests.unit.aws_lambda_app.s3_event import lambda_handler, s3_client
+
+    model.train(hyperparameters={"C": 1.0, "max_iter": 1000})
+    model_path = tmpdir / "model_object.joblib"
+    model.save(str(model_path))
+
+    monkeypatch.setenv("UNIONML_MODEL_PATH", model_path)
+
+    def mock_download_file(bucket, key, filename):
+        # This is a nasty piece of state mutation 😈. We basically
+        # replace the key with the temporary file created in the
+        # lambda handler
+        s3_event["Records"][0]["s3"]["object"]["key"] = filename
+        with Path(filename).open("w") as f:
+            json.dump(features, f)
+
+    def mock_upload_file(filename, bucket, upload_key):
+        ...
+
+    with patch.object(s3_client, "download_file", MagicMock(wraps=mock_download_file)) as mock_dl_file, patch.object(
+        s3_client, "upload_file", MagicMock(wraps=mock_upload_file)
+    ) as mock_ul_file:
+        ret = lambda_handler(s3_event, "")
+
+    assert ret is None
+    mock_dl_file.assert_called_once()
+    mock_ul_file.assert_called_once()
```

Comment:
is there anyway we could capture the content of the file being written to s3? Just to double-check we're writing the right predictions.

Reply:
lemme try something